metaflow/event_logger.py

```diff
@@ -1,33 +1,21 @@
-from .sidecar import SidecarSubProcess
-from .sidecar_messages import Message, MessageTypes
+from metaflow.sidecar import Message, MessageTypes, Sidecar
 
 
-class NullEventLogger(object):
-    def __init__(self, *args, **kwargs):
-        pass
-
-    def start(self):
-        pass
-
+class BaseEventLogger(Sidecar):
     def log(self, payload):
-        pass
+        if self._is_active:
```

Comment:
All methods in `*_logger` and `*_monitor` have this conditional. Is this needed? In what scenario would a logger be assigned but disabled?

Reply:
The Null loggers.